### ISSUE TITLE
setFlyZoneWarningModelCallback and setFlySafeNotificationCallback not work

### ISSUE THREAD

**Issue by vancuong0429** [dji-sdk/Mobile-SDK-Android#1202]
flyZoneManager.setFlySafeNotificationCallback and flyZoneManager.setFlyZoneWarningModelCallback does not call when the drone goes to Restricted Zones
I  use version 4.16.4

**Comment by DJI-William** [comment]
Could you pass me the co-orduiante and your aircraft model. I will test it,

**Comment by vancuong0429** [comment]
@DJI-William My list location(lat,lng):
[(10.836418, 106.688059), (10.838346999999999, 106.693191), (10.828267, 106.69711799999999), (10.799152, 106.619642), (10.802147999999999, 106.61847499999999), (10.800269, 106.613507), (10.81034, 106.60956), (10.839563, 106.686826), (10.808088999999999, 106.61044199999999), (10.808088999999999, 106.61044199999999), (10.800269, 106.613507), (10.802147999999999, 106.61847499999999), (10.799152, 106.619642), (10.783064, 106.59062), (10.786724, 106.58919399999999), (10.78412, 106.58452), (10.803222, 106.57703099999999), (10.81034, 106.60956), (10.851747999999999, 106.717372), (10.854434, 106.722219), (10.835317, 106.729665), (10.828267, 106.69711799999999), (10.830675, 106.69618), (10.830675, 106.69618), (10.838346999999999, 106.693191), (10.836418, 106.688059), (10.839563, 106.686826), (10.855711999999999, 106.715819), (10.801036, 106.577888), (10.801036, 106.577888), (10.78412, 106.58452), (10.786724, 106.58919399999999), (10.783064, 106.59062), (10.745515, 106.522914), (10.749020999999999, 106.521547), (10.746428, 106.51689599999999), (10.786598999999999, 106.501137), (10.803222, 106.57703099999999), (10.889265, 106.78509299999999), (10.891962, 106.78996199999999), (10.851752999999999, 106.80561399999999), (10.835317, 106.729665), (10.854434, 106.722219), (10.851747999999999, 106.717372), (10.855711999999999, 106.715819), (10.893381999999999, 106.78348)]

I use Phantom 4  Pro

**Comment by DJI-William** [comment]
Have you update your flysafe database to the latest. Maybe your flysafe database is not up to date so that the flightController cannot detect the flyzones.

**Comment by DJI-William** [comment]
Yes I tested the same result as yours. The setFlySafeNotificationCallback has no callback. I think this method is broken, I would suggest you use getFlyZonesInSurroundingArea instead.

**Comment by vancuong0429** [comment]
Thanks, Will It be fixed soon?

**Comment by DJI-William** [comment]
Not be soon as the MSDK v5 is our top priority issue to develop. The MSDK v4 is in a low priority.
